File: backend/src/GraphAgentDemo/Models/Project.cs
namespace GraphAgentDemo.Models;

/// <summary>
/// Project status enum
/// </summary>
public enum ProjectStatus
{
    Active,
    Completed,
    OnHold
}

/// <summary>
/// Project entity (adapted from Microsoft Graph API Group resource)
/// </summary>
public class Project
{
    /// <summary>Unique identifier for the project (GUID format)</summary>
    public string Id { get; set; } = string.Empty;
    
    /// <summary>Project name</summary>
    public string Name { get; set; } = string.Empty;
    
    /// <summary>Detailed project description</summary>
    public string Description { get; set; } = string.Empty;
    
    /// <summary>Current project status</summary>
    public string Status { get; set; } = "active";
    
    /// <summary>Project owner (User reference)</summary>
    public User Owner { get; set; } = new User();
    
    /// <summary>When the project was created (ISO 8601 format)</summary>
    public string CreatedDate { get; set; } = DateTime.UtcNow.ToString("o");
    
    /// <summary>Number of team members</summary>
    public int MemberCount { get; set; }
    
    /// <summary>Project email alias (optional)</summary>
    public string? Mail { get; set; }
}

File: frontend/src/types/Project.ts
import type { User } from './User';
import { isValidUser } from './User';

/**
 * Project status enum
 */
export type ProjectStatus = 'active' | 'completed' | 'on-hold';

/**
 * Project entity (adapted from Microsoft Graph API Group resource)
 * @see https://learn.microsoft.com/en-us/graph/api/resources/group
 */
export interface Project {
  /** Unique identifier for the project (GUID format) */
  id: string;
  
  /** Project name */
  name: string;
  
  /** Detailed project description */
  description: string;
  
  /** Current project status */
  status: ProjectStatus;
  
  /** Project owner (User reference) */
  owner: User;
  
  /** When the project was created (ISO 8601 format) */
  createdDate: string;
  
  /** Number of team members */
  memberCount: number;
  
  /** Project email alias (optional) */
  mail?: string;
}

/**
 * Validate project object structure
 * @param project - Object to validate
 * @returns true if valid, false otherwise
 */
export function isValidProject(project: unknown): project is Project {
  const p = project as Record<string, unknown>;
  return (
    typeof p?.id === 'string' &&
    typeof p?.name === 'string' &&
    typeof p?.description === 'string' &&
    ['active', 'completed', 'on-hold'].includes(p?.status as string) &&
    isValidUser(p?.owner) &&
    typeof p?.createdDate === 'string' &&
    typeof p?.memberCount === 'number'
  );
}
